src/kuka_sunrise/configuration_manager.cpp

```diff
@@ -51,13 +73,16 @@ ConfigurationManager::ConfigurationManager(
   if (!robot_manager_node_->has_parameter("receive_multiplier")) {
     robot_manager_node_->declare_parameter("receive_multiplier", rclcpp::ParameterValue(1));
   }
-  if (!robot_manager_node_->has_parameter("controller_ip")) {
-    robot_manager_node_->declare_parameter("controller_ip", rclcpp::ParameterValue("192.168.38.8"));
-  }
-  robot_manager_node_->set_on_parameters_set_callback(
+
+  param_callback_ = robot_manager_node_->add_on_set_parameters_callback(
     [this](
       const std::vector<rclcpp::Parameter> & parameters)
     {return this->onParamChange(parameters);});
+
+  if (!robot_manager_node_->has_parameter("controller_ip")) {
```

Comment:
Why did you put the ip param setting after callback setting?

Reply:
Is it so the onParamChange will be called with the declaration?

Reply:
also, havent you missed setting the actual value?

Reply:
Answer: we need this to be able to change the value from the already set param value.